banditpylib/learners/linear_bandit_learner/linucb.py

```diff
@@ -0,0 +1,127 @@
+from typing import List
+
+import numpy as np
+
+from banditpylib.arms import PseudoArm
+from banditpylib.data_pb2 import Actions, Feedback
+from banditpylib.learners import MaxReward, Goal
+from .utils import LinearBanditLearner
+
+
+class LinUCB(LinearBanditLearner):
+  # incorrect description, to be filled later
+  r"""Upper Confidence Bound policy :cite:`auer2002finite`
+
+    At time :math:`t`, play arm
+
+    .. math::
+          \mathrm{argmax}_{i \in \{0, \dots, N-1\}} \left\{ \bar{\mu}_i(t) +
+          \sqrt{ \frac{\alpha  \ln(t) }{T_i(t)} } \right\}
+    """
+  def __init__(self,
+               arm_num: int,
+               features: List[np.ndarray],
+               delta: float,
+               lambda_reg: float,
+               name: str = None):
+    """
+        Args:
+          arm_num: number of arms
+          name: alias name
+          features: feature vector of each arm in a list
+          delta: delta
+          lambda_reg: lambda for regularization
+        """
+    super().__init__(arm_num=arm_num, name=name)
+    if delta <= 0 or delta >= 1:
+      raise ValueError(
+          'Delta is expected greater than 0 and less than 1. Got %.2f.' %
+          delta)
+
+    if lambda_reg <= 0:
+      raise ValueError('lambda_reg is expected greater than 0. Got %.2f.' %
+                       lambda_reg)
+
+    self.__delta = delta
+    self.__lambda_reg = lambda_reg
+    self.__features = features
+    self.__d = len(features[0])
+    self.__k = len(features)
```

Comment:
Why don't we just use `self.arm_num`?

Reply:
the math formula has k, and hence i assumed it would help improve the readability if a __k variable was used